webapp/src/components/Modal/BidConfirmationModal/BidConfirmationModal.js

```diff
@@ -24,6 +32,60 @@ export default class BidConfirmationModal extends React.PureComponent {
     onAuthorize: PropTypes.func
   }
 
+  constructor(props) {
+    super(props)
+    this.state = {
+      isLoading: false,
+      balance: null
+    }
+  }
+
+  componentWillMount() {
+    const { token, address } = this.props
+    this.fetchBalance(token, address)
+  }
+
+  componentWillReceiveProps({ token, address }) {
+    if (this.props.token !== token || this.props.address !== address) {
+      this.fetchBalance(token, address)
+    }
+  }
+
+  fetchBalance = async (symbol, address) => {
+    this.setState({ balance: null, isLoading: true })
+
+    try {
+      const contractAddress = TOKEN_ADDRESSES[symbol]
+      const response = await axios.get(
+        `https://api.etherscan.io/api?module=account&action=tokenbalance&contractaddress=${contractAddress}&address=${address}&tag=latest&apikey=${env.get(
```

Comment:
shouldn't we check the network here?
Also maybe we can move it to it's own method or object like EtherscanAPI

Reply:
done 👍